src/core/geometry/qgsabstractgeometry.h

```diff
@@ -1026,6 +1033,22 @@ class CORE_EXPORT QgsAbstractGeometry
 
   protected:
 
+    /**
+     * Returns the sort index for the geometry, used in the compareTo() method to compare
+     * geometries of different types.
+     *
+     * \since QGIS 3.20
+     */
+    int sortIndex() const;
+
+    /**
+     * Compares to an \a other geometry of the same class, and returns a integer
+     * for sorting of the two geometries.
```

Comment:
Can you add the information on the returned result, I mean what corresponds to -1, 1, 0?

Reply:
BTW, shouldn't this be an enum?

Reply:
> Can you add the information on the returned result, I mean what corresponds to -1, 1, 0?

I've added a note stating that it's an internal detail only and subject to change between qgis versions :+1:

Reply:
> BTW, shouldn't this be an enum?

No, c++ sorting algorithms always use a int only.